### DISCUSSION TITLE
Difference between output of "colour.read_image" definition and Matlab "imread".

### DISCUSSION THREAD

**Discussion by hminle** [colour-science/colour#738] [Q&A] [answered]
Hi Colour Team,

I using colour.read_image to read a ProPhoto RGB image 16 uint, but as I double check with Matlab imread.
The pixel values are different.

Why is that?

Thanks

**Comment by thomasmansencal** [comment]
Hi @hminle,

Colour converts any image to `float32` by default so I would say this is the core reason for differences: https://github.com/colour-science/colour/blob/c8acc31a8507dae1af81d7958490ac3428dc2728/colour/io/image.py#L246

Cheers,

Thomas

**Comment by thomasmansencal** [comment]
Oh and btw, note that we do not linearise the image, so you would need to know which encoding function was used to save the image so that you can use the corresponding decoding function. If Prophoto RGB it might be [colour.models.cct_decoding_ROMMRGB](https://github.com/colour-science/colour/blob/c8acc31a8507dae1af81d7958490ac3428dc2728/colour/models/rgb/transfer_functions/rimm_romm_rgb.py#L121).

**Comment by hminle** [comment]
> Hi @hminle,
> 
> Colour converts any image to `float32` by default so I would say this is the core reason for differences:
> 
> https://github.com/colour-science/colour/blob/c8acc31a8507dae1af81d7958490ac3428dc2728/colour/io/image.py#L246
> 
> Cheers,
> 
> Thomas

Hi @KelSolaar , even I choose to read image as uint 16 for both Matlab and Colour, it's still different. 
Not sure why?
By the way, when reading into float32, does Colour normalize the output?

Thanks

**Comment by thomasmansencal** [comment, marked as answer]
> By the way, when reading into float32, does Colour normalize the output?

Yes it does! The relevant conversion code is here: https://github.com/colour-science/colour/blob/c8acc31a8507dae1af81d7958490ac3428dc2728/colour/io/image.py#L101

Here is a quick Colab notebook showing some back and forth: https://colab.research.google.com/drive/1D4uB_K2MknkdB6aJ65sejcza8Ex0mKkr

It should be useful! Let me know if you have any questions!

**Comment by thomasmansencal** [comment]
@hminle : Did u get this one working?

**Comment by hminle** [comment]
Hi @KelSolaar, I think the difference lies behind third-party Library, in colour case, it should be ImageIO. Then we cannot tell exactly.

Btw, how could I know "the corresponding decoding function" for each color profile?

**Comment by thomasmansencal** [comment]
> Btw, how could I know "the corresponding decoding function" for each color profile?

It is really hard, depends actually on the writing library, if you know it is using ICC, then via the profile you should get all the needed info, if not, well it is Dark Magic!

**Comment by hminle** [comment]
Hi, so for example, if I load sRGB images using colour, then I manage to convert them to ProPhoto RGB, then I want to save the results. What should I do?

**Comment by thomasmansencal** [comment]
Should be examples to do that in the Colab Notebook above.

**Comment by hminle** [comment]
> Should be examples to do that in the Colab Notebook above.

You mean switch between float 32 and uint16?

**Comment by thomasmansencal** [comment]
They also show how to do  the conversion to ProPhoto RGB.

**Comment by hminle** [comment]
> They also show how to do the conversion to ProPhoto RGB.

Thank you a lot, I see it
